lib/src/cli.rs

```diff
@@ -285,6 +292,46 @@ fn tar_export(opts: &ExportOpts) -> Result<()> {
     Ok(())
 }
 
+// Check if there are any files in a directory.
+fn is_only_directories(path: &Path, error_count: &mut i32) -> Result<bool> {
+    for entry in fs::read_dir(path)? {
+        let entry = entry?;
+        let path = entry.path();
+
+        let metadata = fs::metadata(&path)?;
+
+        if metadata.is_dir() {
+            is_only_directories(&path, error_count)?;
+        } else {
+            *error_count += 1;
+            if error_count < &mut 20 {
+                println!("File {:?} found under '/var' found.", path);
+            }
+        }
+    }
+    Ok(error_count == &mut 0)
+}
+
+async fn container_commit() -> Result<()> {
+    if is_ostree_container().unwrap() {
+        println!("Checking '/var' for files");
+        let var_path = Path::new("/var/");
+
+        let is_dirs = is_only_directories(var_path, &mut 0);
+        if is_dirs.is_err() {
+            return Err(anyhow!(
+                "We encountered an error validating '/var', the error is: {:?}",
+                is_dirs.err()
+            ));
+        } else if !is_dirs.unwrap() {
+            return Err(anyhow!("Many files where found under '/var'. No files are allowed under '/var' just directories."));
+        }
+    } else {
+        println!("Not a container can't commit");
```

Comment:
I think this should actually be an error.  There is `anyhow::bail!` BTW too which is shorthand for `return Err(anyhow!(...));`

Reply:
done